src/olympia/files/models.py

```diff
@@ -274,6 +274,17 @@ def current_file_path(self):
         else:
             return self.file_path
 
+    @property
+    def fallback_file_path(self):
```

Comment:
couldn't we just always try `self.file_path` and then `self.guarded_file_path`.  I'm not sure this extra logic gives us much (apart from one possible attempt at loading a file from the file system which would then fail.

Reply:
I'd rather avoid the extra filesystem calls (that would fail because it's the wrong location) as much as possible. We've got a lot of disabled/deleted files so there is no path that's always best to try first...